src/components/basketPage/basketPage.ts

```diff
@@ -0,0 +1,328 @@
+    this.promoCodes[promo].isUsed = false;
+    code.remove();
+    this.updateTotal();
+  }
+
+  updateTotal(): void {
+    const hasPromo = Object.keys(this.promoCodes).some((keyPromo) => {
+      return this.promoCodes[keyPromo].isUsed;
+    });
+    const promoAmount = document.querySelector('.promo-amount') as HTMLElement;
+    const appliedCodesEl = document.querySelector('.basket-summary__applied-codes') as HTMLElement;
+    const prodAmStart = document.querySelector('#products-amount-start') as HTMLElement;
+    if (hasPromo) {
+      promoAmount.classList.remove('hide');
+      appliedCodesEl.classList.remove('hide');
+      prodAmStart.classList.add('old');
+      let amount = window.app.dataBase.countBasketTotal();
+      promoAmount.innerHTML = `<span>Total: </span>${amount} €`;
+      Object.keys(this.promoCodes).forEach((keyPromo) => {
+        if (this.promoCodes[keyPromo].isUsed) {
+          amount = (amount * (100 - this.promoCodes[keyPromo].percent)) / 100;
+        }
+      });
+      promoAmount.innerHTML = `<span>Total: </span>${amount} €`;
+    } else {
+      promoAmount.classList.add('hide');
+      appliedCodesEl.classList.add('hide');
+      prodAmStart.classList.remove('old');
+    }
+  }
+
+  basketPageItemTemplate(obj: IProducts, index: number, amount: number): string {
+    return `
+      <div class="basket-page__item-num">${index + 1}</div>
+      <div class="basket-page__item-info basket-product-info">
+        <div class="basket-product-info__img">
+          <img src="${obj.thumbnail}" alt="${obj.title}" />
+        </div>
+        <ul class="basket-product-info__details">   
+          <li class="basket-product-info__title">${obj.title}</li>
+          <li class="basket-product-info__category"><span>Category: </span>${obj.category}</li>
+          <li class="basket-product-info__brand"><span>Brand: </span>${obj.brand}</li>
+          <li class="basket-product-info__discount-percent"><span>Discount: </span>${obj.discountPercentage}%</li>
+          <li class="basket-product-info__rating"><span>Rating: </span>${obj.rating}</li>
+          <li class="basket-product-info__stock"><span>Stock: </span>${obj.stock}</li>
+        </ul>
+      </div>
+      <div class="basket-product-info__summary">
+        <div class="basket-product-info__price"><span>Price: </span>${obj.price}&nbsp;€</div>
+        <div class="basket-product-info__amount-info">
+          <button class="basket-product-info__btn btn btn--cart" id="remove-product-btn"> - </button>
+          <span class="basket-product-info__amount amount-of amount-of--dark-bg">${amount}</span>
+          <button class="basket-product-info__btn btn btn--cart" id="add-product-btn"> + </button>
+        </div>
+        <div class="basket-product-info__amount-sum"><span>Total: </span>${obj.price * amount}&nbsp;€</div>
+      </div>
+    `;
+  }
+
+  basketPageTemplate(): string {
+    return `
+      <div class="basket-page">
```

Comment:
this function is pure and static, you can move the markup it returns in some constant variable in some `markupConstants.ts` and import it